### ISSUE TITLE
Infinite block re-confirmation (v13.0)

### ISSUE THREAD

**Issue by justdmitry** [nanocurrency/raiblocks#874]
Logfile full of records "Block ... was confirmed to peers" and "Broadcasting confirm req for block ... to ... representatives" for same block. For several hours.

For example block 3370FDABA82E37C69742827227457630E761861B4A514840E257D03F8BC9DEC1 (logfile lines about this block in attached file).

It looks like *same* block sometimes assumed as confirmed (and being confirmed to peers) and sometimes not (and confirm req is sent).

This seems a little... strange. Node constantly changes it mind about block status.

Also, there is no records in logfile about [changes of] parent of this block.

Why node re-confirms block again and again?

**Additional information:**

> curl -d '{"action":"block", "hash": "3370FDABA82E37C69742827227457630E761861B4A514840E257D03F8BC9DEC1"}' localhost:7076
>> {   "contents": "{\n    \"type\": \"send\",\n    \"previous\": \"5938C1CD3BEF81770E1CCCA153F3BAB58A345F630D225A22DC2AE5FD0B42E517\",\n    \"destination\": \"xrb_3f1r7x6nqyyrwuowb9isp6jrgk5zgrhznaeg183g8m7hs9x9x5hakzdbpgfx\",\n    \"balance\": \"0000000C9F2C9CD04674EDEA3FFFB0E6\",\n    \"work\": \"78a6f78555f18d4b\",\n    \"signature\": \"6E320EC9EC757898959765F8D6A5578FD1B8DCF73E8FD66803BBC58ED88CDA803DE9686362F066A2E7F4D807F5DBEBA5B376A6191BAB790DCAD9F78A438C640B\"\n}\n" }

[History for account xrb_3f1r7x6nqyyrwuowb9isp6jrgk5zgrhznaeg183g8m7hs9x9x5hakzdbpgfx](https://www.nanode.co/account/xrb_3f1r7x6nqyyrwuowb9isp6jrgk5zgrhznaeg183g8m7hs9x9x5hakzdbpgfx) is full of 0-amount receives [from itself](https://www.nanode.co/block/61B64927D66A5CAE3BEE09CE62AA8BF4C699D22EB5C07DCC8DC10CD675B36BFE)! 

**Environment**:

- Ubuntu 16.04 in KVM, no Docker
- `uname -a`: Linux nonna.just-dmitry.ru 4.4.0-124-generic # 148-Ubuntu SMP Wed May 2 13:00:18 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
- rai_nano v13.0

**Logs**

[Part of log related to block 3370FDABA82E37C69742827227457630E761861B4A514840E257D03F8BC9DEC1](https://github.com/nanocurrency/raiblocks/files/2012665/3370FDABA82E37C69742827227457630E761861B4A514840E257D03F8BC9DEC1.log)

**Comment by PlasmaPower** [comment]
For the latter point, PoW is the protection. Since 0 raw is so close to 1 raw, that doesn't really matter (and even if you did you could send 1 nano back and forth between two accounts). We did remove 0 sends in state blocks, but that's more to the fact that we use the difference in balance to determine what the block does, and no difference in ambiguous.

**Comment by justdmitry** [comment]
Agree about 0raw/1raw, closed.

About main thread
In the wake of today's "stresstest"(?) in main network, it seems that problem may be solved by node restarting.

My node was about 50K blocks "behind" (unchecked), generating ~90 log lines per second of "Block XXX was confirmed to peers" (about 5 MB per minute). RPC was unresponsible (5-50 sec per request). After node restart, 35K blocks were checked in about 5 minutes, and node was stuck again. After restart, 14K block were check in next 5 minutes.

But now it with 1630 unconfirmed blocks (9874449 confirmed vs 9874464 on nanode.co) but with "8897 blocks have been unconfirmed averaging 65 announcements" in logs. What are this 8897 blocks while only 1630 are unconfirmed?

**Comment by Onefox** [comment]
I have a smaller case of this:

I use 2 nodes on the testnet, after sending some coins around my first node is spaming the log with:

```
[2018-10-29 13:28:32.728371]: 2 blocks have been unconfirmed averaging 2077 announcements   
[2018-10-29 13:28:32.761403]: Broadcasting confirm req for block 2A3AB5878252593F1A2774B92EB6F2C1E79021697F7E28827F006C3D618ABA49 to 0 representatives                                  
[2018-10-29 13:28:32.761487]: 2 blocks have been unconfirmed averaging 2078 announcements   
[2018-10-29 13:28:32.795287]: Broadcasting confirm req for block 06433E6D423334E01E0A23C9D4FD645560E2489806966143EBF303DF3AFD324D to 0 representatives
[2018-10-29 13:28:32.795455]: 2 blocks have been unconfirmed averaging 2079 announcements
```

Both blocks are already confirmed or at least when I check on both my nodes with the block_confirm call I get a "started": "1" response.

The block ```06433E6D423334E01E0A23C9D4FD645560E2489806966143EBF303DF3AFD324D``` looks like this:

```
{
	"contents": "{
   \"type\": \"state\",
    \"account\": \"xrb_3e3j5tkog48pnny9dmfzj1r16pg8t1e76dz5tmac6iq689wyjfpiij4txtdo\",
    \"previous\": \"04270D7F11C4B2B472F2854C5A59F2A7E84226CE9ED799DE75744BD7D85FC9D9\",
    \"representative\": \"xrb_3e3j5tkog48pnny9dmfzj1r16pg8t1e76dz5tmac6iq689wyjfpiij4txtdo\",
    \"balance\": \"340282366920938463463374607431767211455\",
    \"link\": \"240B07E9F4EEE440463672D2455BB92EE4B33C95A7024AFC11F157A47A2A68C0\",
    \"link_as_account\": \"xrb_1b1d1znzbuq6a355ewpkaofukdq6peybdbr4bdy35wcqnjx4nt81zwpwt4bg\",
    \"signature\": \"BAE13D7F313D11070A36CA2B8A320E7CD8963D4CA25A1D4CE8171E6CF2D6DA53B0F45E1F9E11B9B83D38949256CF92CAA4B788FF3D9D060C7FFB5EECA120B103\",
    \"work\": \"8ad3efe550b3a8ac\"
}
"
}
```

**Comment by rkeene** [comment]
This should be addressed in V17.0 as part of #1254.  Let me know if this issue persists after upgrading to V17.0.  Thanks !
